horace_core/sqw/axes_blocks/@AxesBlockBase/private/bins_in_1Drange_.m

```diff
@@ -18,37 +18,9 @@
 %               considered as being in the range due to the binning algorithm.
 %
 %
-step = (bins(2)-bins(1));
-is_within = false(1,numel(bins));
-ind_min = floor((minmax_val(1)-bins(1))/step)+1;
-ind_max = floor((minmax_val(2)-bins(1))/step)+1;
-if ind_min<1
-    if ind_max<1
-        any_within = false;
-        return;
-    elseif ind_max == 1 % account for partial bin in the beginning
-        if minmax_val(2)<= bins(1) && minmax_val(1) <= bins(1)
-            any_within = false;
-            return;
-        end
-    end
-    ind_min = 1;
-end
-n_bins = numel(bins);
-if ind_max >= n_bins
-    if ind_min>n_bins
-        any_within = false;
-        return;
-    elseif ind_min == n_bins
-        if minmax_val(2)> bins(end) && minmax_val(1)>=bins(end)
-            any_within = false;
-            return;
-        end
-    end
-    % last bin is always false due to the specifics of usage is_within
-    % array for dE binning
-    ind_max  = n_bins-1;
-    if ind_min>ind_max; ind_min=ind_max; end
-end
-any_within = true;
-is_within(ind_min:ind_max) = true;
+wk        = edges<minmax_val(1);
+all_low   = wk(1:end-1)&wk(2:end);
+wk        = edges>minmax_val(2);
+all_high  = wk(1:end-1)&wk(2:end);
+is_within = ~(all_low|all_high);
+any_within= any(is_within);
```

Comment:
Checking that this is the same algorithm as the old code lines 21-54. If not, comments explaining what has been changed would be good. If it is the same, it is a masterly compression of the code, but there are some useful comments in the old code that maybe should be preserved in the new.

Reply:
The algorithm is different and evaluates bin contribution rather then bin edges contribution. The algorithm description is modified to talk about bins rather than edges. 

The commit message says about that and reasons -- difficult to say. It looks like more convenient check in this situation -- part of the total overhaul of the code.